lib/modules/manager/npm/extract/types.ts

```diff
@@ -31,3 +31,9 @@ export interface LockFile {
 export interface PnpmWorkspaceFile {
   packages: string[];
 }
+
+export type OverrideDependency = Record<string, string | RecursiveOverride>;
+
+export interface RecursiveOverride {
+  [P: string]: RecursiveOverride;
+}
```

Comment:
```suggestion
export interface RecursiveOverride  = Record<string, RecursiveOverride>;
```

simplify

Reply:
Unfortunately Typescript doesn't allow recursive types. This can only be done using an interface.